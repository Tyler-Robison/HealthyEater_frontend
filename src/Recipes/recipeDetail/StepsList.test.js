
import React from 'react';
import { render, screen } from '@testing-library/react';
import StepsList from './StepsList';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../../testContext';

const mockInstructions = {
    steps: [
        {
            number: 1,
            step: 'Do the first step',
            equipment: [{ name: 'mock equipment' }, { name: 'other equipment' }],
            ingredients: [{ name: 'mock ingredient' }, { name: 'other ingredient' }]
        },
        {
            number: 2,
            step: 'Do the second step',
            equipment: [],
            ingredients: []
        },
        {
            number: 3,
            step: 'Do the third step',
            equipment: [{ name: 'mock equipment' }],
            ingredients: [{ name: 'mock ingredient' }]
        }
    ]
}

const recipeDetail = {
    analyzedInstructions: [mockInstructions]
}

it("renders without crashing", function () {
    <MemoryRouter>
        <ContextProvider >
            <StepsList recipeDetail={recipeDetail} />
        </ContextProvider>
    </MemoryRouter>
});

it("matches snapshot.", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <StepsList recipeDetail={recipeDetail} />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});