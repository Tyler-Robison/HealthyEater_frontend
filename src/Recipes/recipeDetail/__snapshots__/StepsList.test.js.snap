// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot. 1`] = `
<DocumentFragment>
  <div>
    <div
      class="step-div"
    >
      <p
        class="step-number"
      >
        <b>
          1
        </b>
      </p>
      <p>
        Do the first step
      </p>
      <p>
        <b>
          Equipment Used:
        </b>
         mock equipment, other equipment 
      </p>
      <p>
        <b>
          Ingredients Used:
        </b>
         mock ingredient, other ingredient 
      </p>
    </div>
    <div
      class="step-div"
    >
      <p
        class="step-number"
      >
        <b>
          2
        </b>
      </p>
      <p>
        Do the second step
      </p>
      <p>
        <b>
          Equipment Used:
        </b>
         None 
      </p>
      <p>
        <b>
          Ingredients Used:
        </b>
         None 
      </p>
    </div>
    <div
      class="step-div"
    >
      <p
        class="step-number"
      >
        <b>
          3
        </b>
      </p>
      <p>
        Do the third step
      </p>
      <p>
        <b>
          Equipment Used:
        </b>
         mock equipment 
      </p>
      <p>
        <b>
          Ingredients Used:
        </b>
         mock ingredient 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
